addons/controllers/antreaconfig_controller_test.go

```diff
@@ -227,4 +254,212 @@ var _ = Describe("AntreaConfig Reconciler and Webhooks", func() {
 			Expect(k8sClient.Update(ctx, config)).ToNot(Succeed())
 		})
 	})
+
+	When("antrea config is precreated with {CLUSTER_NAME}-{package-short-name}-package", func() {
+
+		BeforeEach(func() {
+			clusterName = "antrea-custom-cb-cluster"
+			clusterNamespace = "antrea-custom-cb-ns"
+			configName = util.GeneratePackageSecretName(clusterName, constants.AntreaDefaultRefName)
+			tkrString = "v1.23.1"
+			clusterResourceFilePath = "testdata/antrea-custom-tkg-system.yaml"
+			clusterInfraName = "custom-cb-docker-cluster"
+			newDefaultMTU = "8901"
+			ns := &v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: clusterNamespace,
+				},
+			}
+			err := k8sClient.Create(ctx, ns)
+			if err != nil {
+				Expect(apierrors.IsAlreadyExists(err)).To(BeTrue())
+			}
+
+		})
+
+		It("should add cluster as owner reference to antrea config", func() {
+			// define custom resources in cluster namespace
+			f, err := os.Open("testdata/antrea-custom-cb-ns.yaml")
+			Expect(err).ToNot(HaveOccurred())
+			err = testutil.CreateResources(f, cfg, dynamicClient)
+			Expect(err).ToNot(HaveOccurred())
+			f.Close()
+
+			By("Create antrea config in the cluster's namespace with expected pattern", func() {
+				datavalues := &cniv1alpha1.AntreaConfigDataValue{DefaultMTU: newDefaultMTU}
+				antreaConfig := genearateAntreaConfig(configName, clusterNamespace, datavalues)
+				err := k8sClient.Create(ctx, antreaConfig)
+				Expect(err).ToNot(HaveOccurred())
+				err = k8sClient.Get(ctx, client.ObjectKeyFromObject(antreaConfig), antreaConfig)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			By("Create the cluster", func() {
+				cluster := generateDockerCluster(clusterName, clusterNamespace, tkrString, clusterInfraName)
+				err := k8sClient.Create(ctx, cluster)
+				Expect(err).ToNot(HaveOccurred())
+				err = k8sClient.Get(ctx, client.ObjectKeyFromObject(cluster), cluster)
+				Expect(err).ToNot(HaveOccurred())
+
+			})
+
+			By("Verify clusterBootstrap points to pre-created antrea config", func() {
```

Comment:
Verify after reconciliation clusterBootstrap ProviderRef.Name  points to pre-created antrea config